### PR TITLE
Sync with upstream

### PR DESCRIPTION
**Pull request by iafuture**
- Unit tests now pass
- fixed build issues from upstream
- tested with chogori-platform on unit, integration
- validated chogori-platform performance on RDMA machines in lab

### CODE REVIEW

File: src/core/memory.cc [hunk 1]
```diff
@@ -1324,6 +1491,14 @@ void disable_large_allocation_warning() {
 }
 
 void configure(std::vector<resource::memory> m, bool mbind, bool hugepages) {
```
Comment (jfunston) [inline src/core/memory.cc:1493]: With this change I think we should remove the configure calls from our non-seastar SQL threads

**Review by jfunston — COMMENTED** [review]
Did you rebase our commits on top of the current upstream or apply their commits on top of what we had? Were there I conflicts I should review in particular?

File: src/core/memory.cc (hunk 1 above)
Comment (iafuture) [reply src/core/memory.cc:1493]: yeah, i'm working on trying to get SQL to compile

**Comment by iafuture** [comment]
I didn't rebase our commits on top - i merged instead. That way I thought i only have to deal with new upstream commits which touch our commits instead of applying our commits all over again. The particular commits to look into are the last two:

merge conflicts being resolved:
https://github.com/futurewei-cloud/chogori-seastar-rd/pull/5/commits/1591b82b753eef658bf5a4847fc166fe563cee96
and
additional fixes needed to make the codebase build
https://github.com/futurewei-cloud/chogori-seastar-rd/pull/5/commits/3d8a1b89cba658664b7e8cc5602f955b1e4d1a0d

We should go over these together to make sure we didn't miss anything and I can point out some issues i found

**Review by jfunston — APPROVED** [review]

**Pull request merged**